### PR TITLE
Documents events for DisplayObject, renderers, Texture and BaseTexture

### PR DESCRIPTION
**Pull request by bigtimebuddy**
Addresses #3920 

### Added

* Adds additional event documentation including parameters for:
  * Texture
  * BaseTexture
  * DisplayObject
  * WebGLRenderer
  * CanvasRenderer

#### Demo

https://pixijs.download/event-docs/docs/index.html

### CODE REVIEW

**Comment by themoonrat** [comment]
I believe you can do the events as something like
`@event PIXI.WebGLRenderer#postrender`
rather than
```
@event postrender
@memberof PIXI.WebGLRenderer#
```
Just in case you weren't aware. It's personal preference between the cleanliness of an event name on it's own line, or reducing a line used for document an event. I'm not bothered either way.

There are also now some places where we could add the @listens jsdoc, for example, onBaseTextureLoaded for PIXI.BaseTexture#loaded, and onBaseTextureUpdated for PIXI.BaseTexture#updated within Texture.js

**Comment by bigtimebuddy** [comment]
We should do `@fires` and `@listens` for another PR. I mostly wanted to get the missing events documented.

**Comment by darkfrog26** [comment]
@bigtimebuddy it seems there's a ton of missing information regarding events like mouse and keyboard events.  I would expect mouse events to be listed on `DisplayObject`. Is that not a valid assumption?

**Comment by bigtimebuddy** [comment]
Okay, I also added the touch/mouse/pointer events to the DisplayObject (these were documented with the InteractionManager).

Also, there were some missing properties in the documentation like `interactive`, `interactiveChildren` and `hitArea`, all of which are relevant to interaction events.

File: src/interaction/InteractionManager.js [hunk 1]
```diff
@@ -256,210 +256,419 @@ export default class InteractionManager extends EventEmitter
...
+         * @event PIXI.interaction.InteractionManager#pointerup
+         * @param {PIXI.interaction.InteractionData} event - Interaction data
          */
 
         /**
          * Fired when the operating system cancels a pointer event
          *
-         * @event pointercancel
-         * @memberof PIXI.interaction.InteractionManager#
+         * @event PIXI.interaction.InteractionManager#pointercancel
```
Comment (andrewstart) [inline src/interaction/InteractionManager.js]: `pointercancel` and `touchcancel` should have a parameter like the other events.

Comment (bigtimebuddy) [inline src/interaction/InteractionManager.js]: I was wondering about that. Okay, fixed that.

**Comment by bigtimebuddy** [comment]
@darkfrog26 I think I see part of the problem. Namely, that DisplayObject has a bunch of mixins (there's a link at the top of the DisplayObject doc for interactiveTarget). But this is pretty hidden. Properties like `hitArea`, `cursor`, `interactive`, etc were somewhat obscured and definitely weren't showing up in the list on the lift. My recent commit fixes this. 

Also, all the events should now be showing up for all DisplayObject.

**Comment by darkfrog26** [comment]
It would also be beneficial if you'd specify the `originalEvent` type beyond just `Event`.  I presume the `originalEvent` for a mouse `mousedown` would be a `MouseEvent` (https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent)?

**Comment by darkfrog26** [comment]
@bigtimebuddy I haven't checked to verify if this code is different than 4.4.4, but `click` on a `Sprite` doesn't fire an `InteractionData` it fires an `InteractionEvent` instead.

**Comment by bigtimebuddy** [comment]
You are correct @darkfrog26. Thanks for pointing this out. See PR #3923.

**Pull request merged**